io/src/main/java/org/commonjava/maven/ext/io/ModelIO.java

```diff
@@ -186,6 +186,18 @@ public File resolveRawFile( final ArtifactRef ref )
         return versionOverrides;
     }
 
+    public Map<ProjectRef, String> getRemoteDependencyVersionOverridesByProject( final ProjectVersionRef ref )
+                    throws ManipulationException
+    {
+        final Map<ProjectRef, String> versionOverrides = new LinkedHashMap<>();
```

Comment:
Assuming I understand correctly - you need to convert to index off ProjectRef as that will be compared in the override g:a section (?)  wouldn't something like be simpler ? :
```
        return getRemoteDependencyVersionOverrides(ref).entrySet().stream().collect(
                        Collectors.toMap( e -> e.getKey().asProjectRef(), Map.Entry::getValue, (key1, key2) -> {
                            // Avoid problem where a ProjectRef leads to duplicates when the ArtifactRef didn't
                            return key1;
                        } ) );
```

Reply:
That's right. I assume there's some reason the other override maps use ArtifactID keys, but I don't for this, and ProjectIDs allow a simple lookup rather than iterating through a map as a list as `DependencyManipulator.removeGA()` does.

That would be a more succinct conversion, but I haven't worked on Java 8+ code much, so I wrote it in the old lambda-less style.

Reply:
No worries , thanks for changing.